rs/ethereum/cketh/minter/src/eth_logs/mod.rs

```diff
@@ -378,6 +441,30 @@ impl TryFrom<LogEntry> for ReceivedEvent {
     }
 }
 
+fn parse_data_into_32_bytes_words<const N: usize>(
+    data: Data,
+    event_source: EventSource,
+) -> Result<[[u8; 32]; N], ReceivedEventError> {
+    let data = data.0;
+    if data.len() != 32 * N {
+        return Err(ReceivedEventError::InvalidEventSource {
+            source: event_source,
+            error: EventSourceError::InvalidEvent(format!(
+                "Expected {} bytes, got {}",
+                32 * N,
+                data.len()
+            )),
+        });
+    }
+    let mut result = Vec::with_capacity(N);
+    for chunk in data.chunks_exact(32) {
+        let mut word = [0; 32];
+        word.copy_from_slice(chunk);
+        result.push(word);
+    }
+    Ok(result.try_into().unwrap())
```

Comment:
Just for my understanding, if you call `unwrap` anyway, why do you need `try_into`? Is it because `into` doesn't exist?

Reply:
[`TryInto`](https://doc.rust-lang.org/std/convert/trait.TryInto.html) is a standard Rust trait, which is automatically implemented when the source type implements `TryFrom` (blanket implementation). So here, we want to convert a `Vec<T>` (`T` here is `[u8;32]`) into a fixed-size array `[T; N]`), which is obviously a fallible operation (hence it must implement `TryFrom` and not `From`) which is [implemented](https://doc.rust-lang.org/std/primitive.array.html#impl-TryFrom%3CVec%3CT,+A%3E%3E-for-%5BT;+N%5D) by the standard library. We know it shouldn't fail (hence the `unwrap`) because we ensure the correct length of the constructed vector.